tests/it/main.rs

```diff
@@ -44,6 +44,29 @@ fn optionals() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn ensuredir() -> Result<()> {
+    let td = cap_tempfile::tempdir(cap_std::ambient_authority())?;
+
+    let p = Path::new("somedir");
+    let b = &cap_std::fs::DirBuilder::new();
+    assert!(td.metadata_optional(p)?.is_none());
+    assert_eq!(td.ensure_dir_with(p, b).unwrap(), true);
+    assert_eq!(td.ensure_dir_with(p, b).unwrap(), false);
+
+    assert!(td
+        .ensure_dir_with("somedir/without/existing-parent", b)
+        .is_err());
+    assert_eq!(td.ensure_dir_with("somedir/without", b).unwrap(), true);
+    assert_eq!(
+        td.ensure_dir_with("somedir/without/existing-parent", b)
+            .unwrap(),
+        true
+    );
```

Comment:
Instead of `assert_eq!(..., true)`, these could be `assert!(....)`.

Also, this is missing a test for the case where the directory already exists and the function returns `Ok(false)`.

Reply:
There is one of those above, but yeah, might as well do so for the subpath case too.  Done.